File: Planarian/Planarian.Model/Database/Entities/RidgeWalker/PermissionKey.cs
namespace Planarian.Model.Database.Entities.RidgeWalker;

public static class PermissionKey
{
    public const string View = "View";
    public const string Manager = "Manager";
    public const string PlanarianAdmin = "PlanarianAdmin";
    public const string Admin = "Admin";
}

public static class PermissionPolicyKey
{
    public const string View = "View";
    public const string Manager = "Manager";
    public const string PlanarianAdmin = "PlanarianAdmin";
    public const string Admin = "Admin";
    public const string AdminManager = "AdminManager"; // Not actually in the database
    public const string Export = "Export"; // Not actually in the database
}

File: Planarian.Web/src/Modules/Authentication/Models/PermissionKey.ts
export enum PermissionKey {
  View = "View",
  Manager = "Manager",
  AdminManager = "AdminManager",
  Admin = "Admin",
  PlanarianAdmin = "PlanarianAdmin",
  Export = "Export",
}

export const GetPermissionKeyDisplay = (key: PermissionKey): string => {
  switch (key) {
    case PermissionKey.View:
      return "View";
    case PermissionKey.Manager:
      return "Manager";
    case PermissionKey.Admin:
      return "Admin";
    case PermissionKey.PlanarianAdmin:
      return "Planarian Admin";
    default:
      throw new Error("Invalid permission key.");
  }
};
